File: backend/src/main/java/br/ufpr/sept/so2/modules/iam/application/ports/PasswordHasher.java
package br.ufpr.sept.so2.modules.iam.application.ports;

public interface PasswordHasher {

    String hash(String raw);

    boolean matches(String raw, String hash);

    /**
     * Equaliza o tempo de resposta quando o identificador não existe (RNF-SEC-09).
     */
    void matchesDummy();
}

File: backend/src/main/kotlin/br/ufpr/sept/so2/modules/iam/application/ports/PasswordHasher.kt
package br.ufpr.sept.so2.modules.iam.application.ports

interface PasswordHasher {
    fun hash(raw: String): String

    fun matches(raw: String?, hash: String?): Boolean

    /**
     * Equaliza o tempo de resposta quando o identificador não existe (RNF-SEC-09).
     */
    fun matchesDummy()
}
